dissect/target/helpers/configutil.py

```diff
@@ -254,6 +256,59 @@ def parse_file(self, fh: TextIO) -> None:
         self.parsed_data = {"content": fh.read(), "size": str(fh.tell())}
 
 
+class Xml(ConfigurationParser):
+    def _tree(self, tree: ElementTree) -> dict:
+        nodes = {}
+        counter = {}
+        for node in tree.findall("*"):
+            if node.tag in counter:
+                counter[node.tag] += 1
+                nodes[f"{node.tag}-{counter[node.tag]}"] = self._tree(node)
+            else:
+                counter[node.tag] = 1
+                nodes[f"{node.tag}"] = self._tree(node)
+        result = {"tag": tree.tag}
+        if tree.attrib:
+            result["attributes"] = tree.attrib
+        if nodes:
+            result["nodes"] = nodes
+
+        text = str(tree.text).strip(" \n\r")
+        if text:
+            result["text"] = text
+
+        return result
+
+    def _fix(self, content: str, position: tuple(int, int)) -> str:
+        """Quick heuristic fix. If there is an invalid token, just remove it."""
+        lineno, offset = position
+        lines = content.split("\n")
+        line = lines[lineno - 1]
+        line = line[: offset - 1] + "" + line[offset + 1 :]
+        lines[lineno - 1] = line
+        return "\n".join(lines)
+
+    def parse_file(self, fh: TextIO) -> None:
+        content = fh.read()
+        document = content
+        errors = 0
+        limit = 20
+        tree = None
+        while not tree and errors < limit:
+            try:
+                tree = self._tree(ElementTree.fromstring(document))
+                break
+            except ElementTree.ParseError as err:
+                errors += 1
+                document = self._fix(document, err.position)
+
+        if not tree:
+            # Document could not be parsed, we give up
+            self.parsed_data = {"nodes": tree, "errors": errors, "text": content}
+        else:
+            self.parsed_data = {"nodes": tree, "errors": errors}
```

Comment:
```suggestion
        if not tree:
            # Document could not be parsed, we give up
            raise ConfigurationParsingError(f"Could not parse XML file: {fh.name} after {errors} attempts.")

        self.parsed_data = {**tree, "errors": errors}
```

Reply:
We might just want to omit the error count altogether. In that case we can just do `self.parsed_data = tree`